### ISSUE TITLE
RealSR DF2K_JPEG Changes colour space and darkens image

### ISSUE THREAD

**Issue by TechTrooper** [nihui/realsr-ncnn-vulkan#16]
Here's few image comparisons of upscale messing up colors and darkening image:
DVD to 2160p Lanczos vs RealSR
https://imgsli.com/Mjc1MDU

DVD to 2160p Lanczos vs Gigapixel (no issue with colors or darkening)
https://imgsli.com/Mjc1MDY

**Comment by qumaciel** [comment]
I am also having this issue.

**Comment by devernay** [comment]
see https://github.com/Tencent/Real-SR/issues/6 (with a workaround using G'MIC that may work for you)

**Comment by ValZapod** [comment]
It is supposed to be darker because actually you are supposed to show it with 2.4 gamma.

**Comment by TechTrooper** [comment]
> It is supposed to be darker because actually you are supposed to show it with 2.4 gamma.

It makes no sense for the upscaler to change colourspace, black level or gamma of the source material.
Gamma should be changed on the device/player we are using to view (when necessary).

If the encoder darkens the image every time, then after multiple encodings we wouldn't be able to see anything.

**Comment by ValZapod** [comment]
> If the encoder darkens the image every time, then after multiple encodings we wouldn't be able to see anything.

Obviously, like Davinci, you should invert 2.4 gamma and then apply OETF. No loss, besides generation loss (hevc, avc, vp9, av1 all suppport lossless in rgb or ycbcr though).

>It makes no sense for the upscaler to change colourspace, black level or gamma of the source material.

It does not change any data besides metadata. Also, I suppose you do not get it, this is just a bug in ffmpeg. It is supposed to be tagged as 2.4 gamma because otherwise you are just presenting 2.2 gamma on content that is supposed to be 2.4 in dark room. But that would break everything. That is why everone moved to normal EOTF that is just an inverse of OETF, PQ as part of BT.2100.

Oh, and also! The video of DVD does not use the same primaries as HD video. It uses different SMPTE C primaries. And no ONE ever was actually applying linear light conversion AS IT MUST BE and is done on HDR content, BTW.

**Comment by mirh** [comment]
This has nothing to do with ffmpeg, and it's specific to this one neural network.

**Comment by ValZapod** [comment]
What is doing video encode? Did you tag output file as 601 matrix? Since this looks like you did not tag it properly. But I can better if you will use this. https://upload.wikimedia.org/wikipedia/commons/6/60/SMPTE_Color_Bars_16x9.svg

You can immediatley say whether it is 709 as 601 or vice versa.

**Comment by mirh** [comment]
There is no video encode?
Like, you can do whatever you want then with the final images, but you can see the issue even just with

    realsr-ncnn-vulkan -s 4 -t 0 -m models-DF2K_JPEG -g 0 -j 1:1:1 -f png -i 1.png -o 1_output.png

From [this](https://upload.wikimedia.org/wikipedia/commons/thumb/6/60/SMPTE_Color_Bars_16x9.svg/1280px-SMPTE_Color_Bars_16x9.svg.png), you go to this [this](https://user-images.githubusercontent.com/9366725/149551990-b6a1675e-e5f2-4b40-adf1-dbdcc5ab7ba8.png).
